### PR TITLE
Update itemController - update and delete items

### PR DESCRIPTION
**Pull request by junjy**

### CODE REVIEW

File: controllers/ItemController.js [hunk 1]
```diff
@@ -1,18 +1,169 @@
...
+                    images: req.body.images,
+                    delivery: req.body.delivery,
+                    status: req.body.status,
+                    tags: req.body.tags,
+                    postedBy: req.body.postedBy, //to review in frontend
+                };
+
+                for (const prop in itemUpdates) {
+                    if (!itemUpdates[prop]) {
+                        delete itemUpdates[prop];
```
Comment (sumeilee) [inline controllers/ItemController.js:71]: if user updates a field to be an empty string, !itemUpdates[prop] would evaluate to true and the field would be deleted from the update object. This means that any deletion of content won't be updated

File: scripts/seed/itemData.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        title: "Washing Machine (Small)",
+        postType: "Request",
+        description:
+            "Low-income family of 3 living in a 1-room rental flat is in need of a small washing machine (due to the limitation in space) to ease their daily stress over household chores. Father is currently overwhelmed with the caregiving and financial stressors which have arisen from mother (on LTVP)'s recent foot surgery.",
+        category: "Appliances",
+        delivery: "Included",
+        status: "Open",
+        tags: "washing machine, appliances",
+        postedBy: "5fb8b003e1e86a126a37520d",
```
Comment (sumeilee) [inline scripts/seed/itemData.js:11]: not sure why but the postedBy entries in the db are not being converted to ObjectId format...

File: server.js [hunk 3]
```diff
@@ -16,21 +16,37 @@ const mongoURI = `mongodb+srv://${DB_USER}:${DB_PASS}@${DB_HOST}/${DB_NAME}`;
...
+//=== Item Route ====
+
+// // show/get route
+// app.get("/items/:id", itemController.getItem);
+
+// create route
+app.post("/api/v1/items", itemController.createItem);
+
+// update route
+app.patch("/api/v1//items/:id", itemController.updateItem);
```
Comment (sumeilee) [inline server.js:36]: remove extra /

File: server.js [hunk 4]
```diff
@@ -16,21 +16,37 @@ const mongoURI = `mongodb+srv://${DB_USER}:${DB_PASS}@${DB_HOST}/${DB_NAME}`;
...
+
+// delete route
+app.delete("/api/v1//items/:id", itemController.deleteItem);
```
Comment (sumeilee) [inline server.js:39]: remove extra /

**Review by sumeilee — CHANGES_REQUESTED** [review]

File: controllers/ItemController.js (hunk 1 above)
Comment (junjy) [reply controllers/ItemController.js:71]: Updated code to check for empty fields & validate against schema too

File: scripts/seed/itemData.js (hunk 2 above)
Comment (junjy) [reply scripts/seed/itemData.js:11]: Rectified by re-seeding itemData

File: server.js (hunk 3 above)
Comment (junjy) [reply server.js:36]: Updated typo

File: server.js (hunk 4 above)
Comment (junjy) [reply server.js:39]: Updated typo

**Review by sumeilee — DISMISSED** [review]

**Review by sumeilee — APPROVED** [review]

**Pull request merged**